### PR TITLE
Bump numpy version to 1.15.1

### LINKED ISSUE #4451: Builds failing due to new tifffile release
## Description

Example failure: https://travis-ci.org/scikit-image/scikit-image/jobs/651311082#L2729

```pytb
_____________________ ERROR at setup of test_imread_uint16 _____________________
    def setup():
>       use_plugin('tifffile')
../../venv/lib/python3.6/site-packages/skimage/io/tests/test_tifffile.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../venv/lib/python3.6/site-packages/skimage/io/manage_plugins.py:253: in use_plugin
    _load(name)
../../venv/lib/python3.6/site-packages/skimage/io/manage_plugins.py:297: in _load
    fromlist=[modname])
../../venv/lib/python3.6/site-packages/skimage/io/_plugins/tifffile_plugin.py:4: in <module>
    from tifffile import TiffFile, imsave, parse_kwargs
../../venv/lib/python3.6/site-packages/tifffile/__init__.py:3: in <module>
    from .tifffile import __doc__, __all__, __version__, main
../../venv/lib/python3.6/site-packages/tifffile/tifffile.py:524: in <module>
    import imagecodecs
../../venv/lib/python3.6/site-packages/imagecodecs/__init__.py:5: in <module>
    from .imagecodecs import __doc__, __version__, __getattr__, __dir__
../../venv/lib/python3.6/site-packages/imagecodecs/imagecodecs.py:804: in <module>
    _load_all()
../../venv/lib/python3.6/site-packages/imagecodecs/imagecodecs.py:347: in _load_all
    __getattr__(name)
../../venv/lib/python3.6/site-packages/imagecodecs/imagecodecs.py:371: in __getattr__
    module = importlib.import_module('._' + module_, 'imagecodecs')
../../venv/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
>   ???
E   AttributeError: type object 'imagecodecs._aec.array' has no attribute '__reduce_cython__'
```

I forget where the `has no attribute '__reduce_cython__'` error comes from but we've seen it before...

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Closes #4451

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
I'm not sure we are quite ready for this.

xref: https://github.com/scikit-image/scikit-image/issues/4259

**Comment by rfezzani** [comment]
Sure, I saw your comment ;-) We are ready in case...

**Comment by rfezzani** [comment]
The to-do list was wrong concerning [`arraycrop`](https://github.com/scikit-image/scikit-image/pull/4452/files#diff-ddc4bbaedfeb2dbe635f7eb7907b98d0L67) it caused the CI failures :sob: 
I will fix it and revert these changes

**Comment by jni** [comment]
@rfezzani looks like there's some warnings not raised now that are causing the test failures...

File: skimage/filters/tests/test_lpi_filter.py [hunk 1]
```diff
@@ -1,20 +1,10 @@
...
-from distutils.version import LooseVersion as Version
 import unittest
 
 from skimage import data
 from skimage.filters import LPIFilter2D, inverse, wiener
-from skimage._shared._warnings import expected_warnings
-
-
-if (Version(np.__version__) >= '1.15.0' and
-        Version(scipy.__version__) <= '1.1.0'):
```
Comment (soupault) [inline skimage/filters/tests/test_lpi_filter.py]: I think scipy version check should be kept.

File: skimage/segmentation/tests/test_random_walker.py [hunk 2]
```diff
@@ -14,10 +12,7 @@
 PYAMG_MISSING_WARNING = r'pyamg|\A\Z'
 PYAMG_OR_SCIPY_WARNING = SCIPY_RANK_WARNING + '|' + PYAMG_MISSING_WARNING
 
-if Version(np.__version__) >= '1.15.0' and Version(scipy.__version__) < '1.3':
```
Comment (soupault) [inline skimage/segmentation/tests/test_random_walker.py]: Here also.

**Comment by rfezzani** [comment]
:tada: Halleluja! Finally  got rid of the  CI failures!!!

File: skimage/filters/tests/test_lpi_filter.py [hunk 3]
```diff
@@ -10,11 +10,7 @@
 from skimage._shared._warnings import expected_warnings
 
 
-if (Version(np.__version__) >= '1.15.0' and
-        Version(scipy.__version__) <= '1.1.0'):
-    SCIPY_ND_INDEXING_WARNING = 'non-tuple sequence for multidimensional'
-else:
-    SCIPY_ND_INDEXING_WARNING = None
+SCIPY_ND_INDEXING_WARNING = None
```
Comment (hmaarrfk) [inline skimage/filters/tests/test_lpi_filter.py]: Want to just remove this all together?

Comment (rfezzani) [reply skimage/filters/tests/test_lpi_filter.py]: Yes, it is ready, just wanted to see the CI full green before pushing :stuck_out_tongue_closed_eyes:

Comment (rfezzani) [reply skimage/filters/tests/test_lpi_filter.py]: OK, can't wait :smile:

**Comment by hmaarrfk** [comment]
wow awesome!

**Comment by jni** [comment]
Since this is maintenance code, follows NEP-29, and gets our CI green again, imma go ahead and merge! =D

**Comment by rfezzani** [comment]
Thank you @jni.

**Pull request merged**